pulpcore/cli/common/generic.py

```diff
@@ -360,3 +368,38 @@ def version_group(
 version_group.add_command(show_version)
 version_group.add_command(destroy_version)
 version_group.add_command(repair_version)
+
+
+# Generic label command group
+@click.group(name="label")
+@pass_entity_context
+@pass_pulp_context
+@click.pass_context
+def label_group(ctx: click.Context, entity_ctx: PulpEntityContext, pulp_ctx: PulpContext) -> None:
+    """Manage labels for this resource"""
+    pass
+
+
+@label_group.command(name="set")
+@click.option("--name", required=True, help="Name of the entry")
```

Comment:
I think, the new factory pattern applies quite nicely to those commands. You can move the lookup question into the hands of the code using the factory.
I managed to make some file commands able to understand both name and href that way.